tools/circt-verilog/circt-verilog.cpp

```diff
@@ -16,22 +16,293 @@
+
+  cl::opt<bool> onlyParse{
+      "parse-only",
+      cl::desc(
+          "Only parse and elaborate the input, without mapping to CIRCT IR"),
+      cl::init(false), cl::cat(cat)};
+
+  //===--------------------------------------------------------------------===//
+  // Include paths
+  //===--------------------------------------------------------------------===//
+
+  cl::list<std::string> includeDirs{
+      "I", cl::desc("Additional include search paths"), cl::value_desc("dir"),
+      cl::Prefix, cl::cat(cat)};
+  cl::alias includeDirsLong{"include-dir", cl::desc("Alias for -I"),
+                            cl::aliasopt(includeDirs), cl::NotHidden,
+                            cl::cat(cat)};
+
+  cl::list<std::string> includeSystemDirs{
+      "isystem", cl::desc("Additional system include search paths"),
+      cl::value_desc("dir"), cl::cat(cat)};
+
+  cl::list<std::string> libDirs{
+      "y",
+      cl::desc(
+          "Library search paths, which will be searched for missing modules"),
+      cl::value_desc("dir"), cl::Prefix, cl::cat(cat)};
+  cl::alias libDirsLong{"libdir", cl::desc("Alias for -y"),
+                        cl::aliasopt(libDirs), cl::NotHidden, cl::cat(cat)};
+
+  cl::list<std::string> libExts{
+      "Y", cl::desc("Additional library file extensions to search"),
+      cl::value_desc("ext"), cl::Prefix, cl::cat(cat)};
+  cl::alias libExtsLong{"libext", cl::desc("Alias for -Y"),
+                        cl::aliasopt(libExts), cl::NotHidden, cl::cat(cat)};
+
+  cl::list<std::string> excludeExts{
+      "exclude-ext",
+      cl::desc("Exclude provided source files with these extensions"),
+      cl::value_desc("ext"), cl::cat(cat)};
+
+  //===--------------------------------------------------------------------===//
+  // Preprocessor
+  //===--------------------------------------------------------------------===//
+
+  cl::list<std::string> defines{
+      "D",
+      cl::desc("Define <macro> to <value> (or 1 if <value> ommitted) in all "
+               "source files"),
+      cl::value_desc("<macro>=<value>"), cl::Prefix, cl::cat(cat)};
+  cl::alias definesLong{"define-macro", cl::desc("Alias for -D"),
+                        cl::aliasopt(defines), cl::NotHidden, cl::cat(cat)};
+
+  cl::list<std::string> undefines{
+      "U", cl::desc("Undefine macro name at the start of all source files"),
+      cl::value_desc("macro"), cl::Prefix, cl::cat(cat)};
+  cl::alias undefinesLong{"undefine-macro", cl::desc("Alias for -U"),
+                          cl::aliasopt(undefines), cl::NotHidden, cl::cat(cat)};
+
+  cl::opt<uint32_t> maxIncludeDepth{
+      "max-include-depth",
+      cl::desc("Maximum depth of nested include files allowed"),
+      cl::value_desc("depth"), cl::cat(cat)};
+
+  cl::opt<bool> librariesInheritMacros{
+      "libraries-inherit-macros",
+      cl::desc(
+          "If true, library files will inherit macro definitions from the "
+          "primary "
+          "source "
```

Comment:
Wish clang-format could re-join strings at least sometimes, maybe want to tweak this?

Reply:
Ooops, yeah